internal/config/config.go

```diff
@@ -0,0 +1,218 @@
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"sigs.k8s.io/kubebuilder/pkg/scaffold/resource"
+	"sigs.k8s.io/yaml"
+)
+
+// constants for scaffolding version
+const (
+	Version1 = "1"
+	Version2 = "2"
+
+	DefaultPath = "PROJECT"
+)
+
+// Config is the unmarshalled representation of the configuration file
+type Config struct {
+	// Path stores where the config was loaded from
+	path string
+
+	// Version is the project version, defaults to "1" (backwards compatibility)
+	Version string `json:"version,omitempty"`
+
+	// Domain is the domain associated with the project and used for API groups
+	Domain string `json:"domain,omitempty"`
+
+	// Repo is the go package name of the project root
+	Repo string `json:"repo,omitempty"`
+
+	// Resources tracks scaffolded resources in the project
+	// This info is tracked only in project with version 2
+	Resources []GVK `json:"resources,omitempty"`
+
+	// Multigroup tracks if the project has more than one group
+	MultiGroup bool `json:"multigroup,omitempty"`
+}
+
+func exists(path string) (bool, error) {
+	// Look up the file
+	_, err := os.Stat(path)
+
+	// If we could find it the file exists
+	if err == nil {
+		return true, nil
+	}
+
+	// Not existing and different errors are differentiated
+	if os.IsNotExist(err) {
+		err = nil
+	}
+	return false, err
+}
+
+// Exists verifies that the configuration file exists in the default path
+func Exists() (bool, error) {
+	return exists(DefaultPath)
+}
+
+// ExistsIn verifies that the configuration file exists in the provided path
+func ExistsIn(path string) (bool, error) {
+	return exists(path)
+}
+
+func load(path string) (*Config, error) {
+	// Read the file
+	in, err := ioutil.ReadFile(path) // nolint: gosec
+	if err != nil {
+		return &Config{}, err
+	}
+
+	// Unmarshal the file content
+	config := &Config{}
+	if err = yaml.Unmarshal(in, config); err != nil {
+		return &Config{}, err
+	}
+
+	// Store where it was loaded from
+	config.path = path
+
+	// kubebuilder v1 omitted version, so default to v1
+	if config.Version == "" {
+		config.Version = Version1
+	}
+
+	return config, nil
+}
+
+// Load obtains the configuration from the default path
+func Load() (*Config, error) {
+	return load(DefaultPath)
+}
+
+// LoadFrom obtains the configuration from the provided path
+func LoadFrom(path string) (*Config, error) {
+	return load(path)
+}
+
+func (config Config) save(path string) error {
+	content, err := yaml.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("error marshalling project configuration: %v", err)
+	}
+
+	err = ioutil.WriteFile(path, content, os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("failed to save configuration to %s: %v", path, err)
+	}
+
+	return nil
+}
+
+// Save saves the configuration information
+func (config Config) Save() error {
+	return config.save(config.path)
+}
+
+// CopyTo saves a copy of the configuration information to the provided path
+func (config Config) CopyTo(path string) error {
+	return config.save(path)
+}
```

Comment:
I do not think that the CopyTo is required,

Reply:
It is unused, I added it for completeness. The same way we have `Load` and `LoadFrom` (which will be renames to `NewConfig` and `NewConfigFrom`), I added `Save` and `CopyTo`. I actually plan to use it in a near future to scaffold this file, as we do not need all the scaffolding. I can leave it unexported for now with a note.

Reply:
Removed